observer/probers/dns/dns_conf.go

```diff
@@ -89,10 +91,17 @@ func (c DNSConf) validateQType() error {
 		"invalid `query_type`, got: %q, expected one in %s", c.QType, q)
 }
 
+func (c DNSConf) validateCollectors(colls map[string]*prometheus.Collector) error {
+	if colls != nil {
+		return errors.New("dns prober defines no metrics but received non-nil collector map")
+	}
+	return nil
+}
+
 // MakeProber constructs a `DNSProbe` object from the contents of the
 // bound `DNSConf` object. If the `DNSConf` cannot be validated, an
 // error appropriate for end-user consumption is returned instead.
-func (c DNSConf) MakeProber() (probers.Prober, error) {
+func (c DNSConf) MakeProber(colls map[string]*prometheus.Collector) (probers.Prober, error) {
```

Comment:
It think it makes sense to just write this var name out in it's complete form.

```suggestion
func (c DNSConf) MakeProber(collectors map[string]*prometheus.Collector) (probers.Prober, error) {
```